### PR TITLE
feat(runner): separate control and per-run data networks

### PR DESCRIPTION
**Pull request by Stebalien**
1. Use separate control/data networks everywhere.
2. Create a new data network for every run.

I've tested the local docker runner changes but am unsure how to test it on AWS.

### CODE REVIEW

File: pkg/runner/cluster_swarm.go [hunk 1]
```diff
@@ -155,12 +155,47 @@ func (*ClusterSwarmRunner) Run(input *api.RunInput) (*api.RunOutput, error) {
...
+		Driver:         "overlay",
+		CheckDuplicate: true,
+		EnableIPv6:     true, // TODO: params?
+		Internal:       true,
+		Attachable:     true,
+		Scope:          "swarm",
+		Labels: map[string]string{
+			"testground.plan":     input.TestPlan.Name,
+			"testground.testcase": testcase.Name,
+			"testground.runid":    input.RunID,
```
Comment (Stebalien) [inline pkg/runner/cluster_swarm.go]: I'm using this to find all networks associated with the given test run.

File: pkg/runner/cluster_swarm.go [hunk 2]
```diff
@@ -155,12 +155,47 @@ func (*ClusterSwarmRunner) Run(input *api.RunInput) (*api.RunOutput, error) {
...
+			"testground.name":     "default", // default name. TODO: allow multiple networks.
```
Comment (Stebalien) [inline pkg/runner/cluster_swarm.go]: This is the name that tests and the sidecar will use. (we may support more network interfaces in the future)

File: pkg/runner/cluster_swarm.go [hunk 3]
```diff
@@ -172,6 +207,11 @@ func (*ClusterSwarmRunner) Run(input *api.RunInput) (*api.RunOutput, error) {
 			ContainerSpec: &swarm.ContainerSpec{
 				Image: image,
 				Env:   env,
+				Labels: map[string]string{
+					"testground.plan":     input.TestPlan.Name,
```
Comment (raulk) [inline pkg/runner/cluster_swarm.go]: Nice one. These labels are gonna come in handy in general.

**Comment by raulk** [comment]
Fixes #153.

**Comment by Stebalien** [comment]
Thanks for looking into that.

**Pull request merged**